app/credentials/oauth/oauth.go

```diff
@@ -42,27 +38,30 @@ func NewCredential(accessToken string, opts ...Option) (Credential, error) {
 }
 
 func (c Credential) GetRequestMetadata(ctx context.Context, uri ...string) (map[string]string, error) {
-	ri, ok := credentials.RequestInfoFromContext(ctx)
+	ri, ok := grpccreds.RequestInfoFromContext(ctx)
 	if !ok {
 		return nil, fmt.Errorf("failed to retrieve request info from context")
 	}
 
 	if !c.allowInsecureTransport {
 		// Ensure the bearer token is sent over a secure connection - meaning TLS.
-		if err := credentials.CheckSecurityLevel(ri.AuthInfo, credentials.PrivacyAndIntegrity); err != nil {
-			return nil, fmt.Errorf("the connection's transport security level is too low for OAuth: %v", err)
+		if err := grpccreds.CheckSecurityLevel(ri.AuthInfo, grpccreds.PrivacyAndIntegrity); err != nil {
+			return nil, fmt.Errorf("the connection's transport security level is too low for OAuth: %w", err)
 		}
 	}
 
+	token, err := c.source.Token()
+	if err != nil {
+		return nil, fmt.Errorf("failed to retrieve token from token source: %w", err)
+	}
+
 	return map[string]string{
-		Header: c.token(),
+		credentials.AuthorizationHeader: fmt.Sprintf("%s %s", token.Type(), token.AccessToken),
 	}, nil
 }
 
 func (c Credential) RequireTransportSecurity() bool {
 	return !c.allowInsecureTransport
 }
 
-func (c Credential) token() string {
-	return "Bearer " + c.accessToken
-}
+var _ grpccreds.PerRPCCredentials = (*Credential)(nil)
```

Comment:
noob q: what does this do?

Reply:
Its a good question. It ensures that the structure `Credential` that we wrote satisfies the `grpccreds.PerRPCCredentials` interface at compile time. Its called an [interface check](https://go.dev/doc/effective_go#blank_implements) in Go, and is a consequence of Go interfaces being implicit.

They made them implicit (meaning no `implements` keyword) to decouple the definition of an interface from the implementation. It also allows a structure to implement many interfaces, and encourages a design idiom in Go where interfaces are small and well-defined. And other things.

Great blog post on [why interfaces are implicit](https://dave.cheney.net/2016/08/20/solid-go-design) from Dave Cheney.